lib/compat.php

```diff
@@ -24,18 +24,90 @@
 	 * @return WP_Block_Type|false The registered block type on success, or false on failure.
 	 */
 	function register_block_type_from_metadata( $path, $args = array() ) {
-		$file = trailingslashit( $path ) . 'block.json';
-		if ( ! file_exists( $file ) ) {
+		$metadata_file = trailingslashit( $path ) . 'block.json';
+		if ( ! file_exists( $metadata_file ) ) {
 			return false;
 		}
 
-		$metadata = json_decode( file_get_contents( $file ), true );
-		if ( ! is_array( $metadata ) ) {
+		$metadata = json_decode( file_get_contents( $metadata_file ), true );
+		if ( ! is_array( $metadata ) || ! $metadata['name'] ) {
 			return false;
 		}
 
+		$block_dir           = dirname( $metadata_file );
+		$block_name          = $metadata['name'];
+		$asset_handle_prefix = str_replace( '/', '-', $block_name );
+
+		if ( isset( $metadata['editorScript'] ) ) {
+			$editor_script            = $metadata['editorScript'];
+			$editor_script_handle     = "$asset_handle_prefix-editor-script";
+			$editor_script_asset_path = realpath(
+				$block_dir . '/' . substr_replace( $editor_script, '.asset.php', -3 )
```

Comment:
To the point in my previous comment about utility functions, even just for the purpose of self-documenting the implementation, it would help to clarify what's happening here with `substr_replace( ..., -3 )`

Reply:
I guess:
```js
substr_replace( $editor_script, '.asset.php', - strlen( '.js' ) )
```
would read better but still I get your point :)